update sphinx docs

karosc opened the pull request:
This PR updates the sphinx docs to align with the style of the new pyswmm website:
- correct existing doc string syntax to python3 where needed
- use pydata_sphinx_theme
- add new pyswmm logo to site
- new landing page with sphinx-panels

Preview docs at: [https://www.karosc.com/pyswmm/index.html](https://www.karosc.com/pyswmm/index.html)

bemcdonnell commented:
@karosc great work in this!!! Quick question, do you have the ability to choose how objects module docs are sorted? For example, in links I’d like to see 

Links
Link
Conduit (and example code would be good here) 
Pump
…. 

I don’t mean to be too particular. It’s just that Links/Link are heavily discussed and in our docs and tutorials at the moment. I am hoping it is just intuitive for people

karosc commented:
@bemcdonnell, we can totally change the order by explicitly writing autoclass rst or simply by asking sphinx automodule to order member bysource rather than alphabetically. 

I also wanted to ask you about whether you'd like to include inherited methods and props for each class. For example, the Conduit class inherits from Link, but the docs only show the methods written in the Conduit class, requiring the user to look through the Link doc for other methods inherited by Conduit. We can change the rst config to automatically include inherited methods and properties for each child class as well as parent class. I personally think it is easier to see all available methods to a class in its own doc, no matter if they are inherited or not. Thoughts?

bemcdonnell commented:
@karosc, I think the inheritance should be included (but at the bottom). In the doc strings we can add how to access those via the Link.  This is my opinion but I trust your judgement!

do you have links to the website and YouTube videos?

karosc commented:
@bemcdonnell , so for example, the Conduit class only has one additional property `conduit_statistics`. You would want that property at the top of the Conduit class doc, then all the inherited Link methods and properties below that. Correct? Order doesn't matter that much to me....as long as all the available methods to a class are presented in that class's doc, I'll be happy.

I set the examples link in the top bar to link back to pyswmm.org/examples for now. Eventually it'd be nice to integrate those examples into the doc site, potentially as a [cloud executable binder](https://mybinder.org/) or something similar with the youtube video embedded on the page.

Other than the one that leads to the examples, I don't have a link to the website yet. However, we can add a pyswmm logo icon next to the github and pypi logos that links to pyswmm.org. For something less discrete, we can also add a larger link on the overview page, or on the landing page.

I also want to chat with you about the readthedocs config and the new domain. The updates docs have some build dependencies that we will need to config for readthedocs, hoping we can go through that on slack when you have time. Also, now that we have pyswmm.org, we could [setup a subdomain](https://docs.readthedocs.io/en/stable/custom-domains.html) that could link to the documentation (e.g. docs.pyswmm.org).

bemcdonnell commented:
@karosc I'm thinking this order:

Links
Link
Conduits (add example)
Pumps (add example)

I can totally add docs.pyswmm.org to the subdomain! Just give me a URL to hit and I got it.  We are using google sites now... not completely satisfied but it is getting the job done.  Thinking we need a space for blogging one day (not high pressure at the moment).  We can redirect to the RTD page for now if you like?  

Can you add some details /links to the website in the home page?

karosc commented:
@bemcdonnell I decided not to add inherited methods. After doing it locally, I found that it made the page quite long and harder to navigate.....not to mention there is no concise way to get sphinx to push inherited members to the bottom. Nevertheless, [I made a few updates](https://www.karosc.com/pyswmm/):

1. added the pyswmm badges and a link to the office website to the home page
2. added pyswmm logo to to right as a link to pyswmm.org
3. added example code to Conduit and Pump class methods
4. fixed minor typos in a few docstrings

karosc commented:
@bemcdonnell and @abhiramm7 , I revised the docs to use autosummary for the api reference. I think it provides a better way of navigating the methods and properties of each class. Let me know what you think.

[www.karosc.com/pyswmm](www.karosc.com/pyswmm)

abhiramm7 commented:
This looks beautiful 😍. I think we should do a minor release with this addition.

abhiramm7 commented on .gitignore:
```diff
@@ -39,3 +39,6 @@ labels.txt
 *.thm
+
+#IDE Folders
+.vscode/
```
It is a personal opinion; please feel free to ignore it. I think we should try to have a gitignore that only houses package-specific things.

bemcdonnell commented:
@karosc, take a look at matplotlib for one other idea:  

https://matplotlib.org/stable/api/cbook_api.html

if you notice, there is a little button over the code where you can "copy."  If you use it, it removes all the `>>>` lines and keeps the good formatting.  I was trying to dive into Sphinx more to see how to enable this and I believe it is just as simple as  tabbing in the code region.  However, I am no expert in Sphinx.  

I just have couple more ideas:
 - [ ] Can you add the PySWMM artistic banner to the landing AND overview pages? 
 - [ ] I'm not sure if this is an option for sphinx but the PySWMM Website uses "Avenir" font
 - [ ] I wonder if there are additional icons that can be added for the YouTube and LinkedIN spaces here:
<img width="260" alt="image" src="https://user-images.githubusercontent.com/6105134/213917835-21e8e06f-51ab-4d2e-94ef-699f7c932a5b.png">

I think the PR looks good and I will leave it up to you as to whether you agree with these proposed ideas! :) 

GREAT WORK!!!

bemcdonnell reviewed (approved):
See my recent comment!  Looks really good! Nice work!

bemcdonnell commented:
@karosc last comment!

https://www.karosc.com/pyswmm/install.html

let's make sure the docs say you need 64bit Python :)

bemcdonnell commented:
Other than that, we are green light! these look so awesome.  I'm thankful for you time and energy into this.  It means the world to the project and all our new user! @karosc for president!

karosc commented:
> let's make sure the docs say you need 64bit Python :)

should we add 32-bit support in this release?
